src/librustc/lint.rs

```diff
@@ -19,10 +19,10 @@ pub enum LintSource {
     Default,
 
     /// Lint level was set by an attribute.
-    Node(Symbol, Span, Option<Symbol> /* RFC 2383 reason */),
+    Node(Symbol, Option<Level>, Span, Option<Symbol> /* RFC 2383 reason */),
```

Comment:
Perhaps this (and the `CommandLine` variant) should be made into struct variants with named fields for clarity?